tests/ui/single_match.fixed

```diff
@@ -296,3 +296,21 @@ fn issue11365() {
 
     if let Some(A | B) = &Some(A) { println!() }
 }
+
+#[derive(Eq, PartialEq)]
+pub struct Data([u8; 4]);
+
+const DATA: Data = Data([1, 2, 3, 4]);
+const CONST_I32: i32 = 1;
+
+fn irrefutable_match() {
+    { println!() }
+
+    { println!() }
```

Comment:
Can we suggest this into an expr instead of a block?

For L313, I guess it might change the behaviour if we move out of the block, but this part is fine.